presto-main/src/test/java/io/prestosql/metadata/TestInformationSchemaConnector.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.metadata;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.Session;
+import io.prestosql.connector.MockConnectorFactory;
+import io.prestosql.plugin.tpch.TpchConnectorFactory;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.testing.LocalQueryRunner;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestInformationSchemaConnector
+{
+    private LocalQueryRunner queryRunner;
```

Comment:
Why don't you `extends AbstractTestQueryFramework`?
This would take care of `queryRunner` set up & disposal and would provide methods for testing query results.

Reply:
`AbstractTestQueryFramework` is in module `presto-tests`. When adding that dependency, Intellij gives a warning:
```
Circular Dependency Warning
Adding dependency on module 'presto-tests' will introduce circular dependency between modules 'presto-main' and 'presto-resource-group-managers'. Add dependency anyway?
```
I suppose it's not okay to do so?

Reply:
@linxingyuan1102 good point. I'd move `TestInformationSchemaConnector.java` to `presto-tests` unless @kokosing says the contrary.

Reply:
Would moving `AbstractTestQueryFramework` to `presto-main` tests help here?

Reply:
@kokosing 
I tried and found that moving `AbstractTestQueryFramework` to `presto-main` involves much more code change because of transitive dependencies.

Reply:
@findepi @kokosing 
I found that we cannot utilize query verification methods in `AbstractTestQueryFramework`, e.g., `assertQuery(@Language("SQL") String sql)`. Because the semantics of information_schema seem to be different for Presto and H2. For the same query, `select * from information_schema.tables`, Presto returns 80 records while H2 returns 38 records.
Do we still want to `extends AbstractTestQueryFramework`?

Reply:
i missed that, sorry. You can send a different query to h2, which will return the expected results. Often, we use `VALUES ...` queries.
Of course, this works only when query returns reasonable amount of data, which isn't the case here in some cases.

To consider:
- move the test to presto-tests
- extend `AbstractTestQueryFramework`
- use `assertQuery(String, String)` where it makes sense 
- use `computeActual` + manual verification where `assertQuery` doesn't make sense, e.g. would be overly verbose.

Reply:
Thanks for the pointers. I've updated `TestInformationSchemaConnector.java`. I also extracted it to be an independent commit so that we can be sure that it works with implementations before and after a change.